bundles/org.openhab.binding.unifi/src/main/java/org/openhab/binding/unifi/internal/api/cache/UniFiCache.java

```diff
@@ -29,41 +33,65 @@
  * <code>prefix:suffix</code> are searched in the order of their priority.
  *
  * @author Matthew Bowman - Initial contribution
+ * @author Hilbrand Bouwkamp - Moved generic code into this class
  */
-public abstract class UniFiCache<T> {
+@NonNullByDefault
+abstract class UniFiCache<T extends @Nullable HasId> {
 
     private static final String SEPARATOR = ":";
 
     public static final String PREFIX_ALIAS = "alias";
-
     public static final String PREFIX_DESC = "desc";
-
     public static final String PREFIX_HOSTNAME = "hostname";
-
     public static final String PREFIX_ID = "id";
-
     public static final String PREFIX_IP = "ip";
-
     public static final String PREFIX_MAC = "mac";
-
     public static final String PREFIX_NAME = "name";
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
+    // Map of cid keys to the id.
+    private final Map<String, String> mapToId = new HashMap<>();
+    // Map of id to data object
+    private final Map<String, T> map = new HashMap<>();
+    private final String[] prefixes;
+    private final String type;
+
+    protected UniFiCache(final String type, final String... prefixes) {
```

Comment:
I would suggest to introduce an enum for type instead of using strings.

Reply:
@Hilbrand - maybe some misunderstanding when looking at latest commit: Was suggesting for **type**, i.e. to avoid hardcoded strings like "Client", "Site", "Device" etc. as first constructor argument.